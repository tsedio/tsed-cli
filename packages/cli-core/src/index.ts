import "./utils/patchCommander.js";

import Inquirer from "inquirer";

export * from "./CliCore.js";
export * from "./decorators/index.js";
export * from "./interfaces/index.js";
export * from "./packageManagers/index.js";
export * from "./services/index.js";
export * from "./utils/index.js";
export * from "@tsed/core";
export * from "@tsed/di";
export * from "@tsed/logger";
export * from "@tsed/normalize-path";
export {Inquirer};
