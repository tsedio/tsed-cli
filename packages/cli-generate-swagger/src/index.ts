export * from "./commands/GenerateSwaggerCmd";

export default {};
