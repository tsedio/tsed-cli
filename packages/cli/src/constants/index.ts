import {dirname, join} from "node:path";

import {readPackageUpSync} from "read-pkg-up";

const {path, packageJson} = readPackageUpSync({
  cwd: join(import.meta.dirname, "..", "..")
})!;

export const PKG = packageJson;
export const MINIMAL_TSED_VERSION = "8";
export const DEFAULT_TSED_TAGS = "latest";
export const IGNORE_VERSIONS = ["6.0.0"];
export const IGNORE_TAGS: false | RegExp = false; // /alpha|beta/
export const TEMPLATE_DIR = join(dirname(path), "templates");
