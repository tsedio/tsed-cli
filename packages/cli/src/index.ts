import "./templates/index.js";

export * from "./Cli.js";
export * from "./commands/add/AddCmd.js";
export * from "./commands/generate/GenerateCmd.js";
export * from "./commands/init/config/FeaturesPrompt.js";
export * from "./commands/init/config/FeaturesPrompt.js";
export * from "./commands/init/InitCmd.js";
export * from "./commands/init/prompts/getFeaturesPrompt.js";
export * from "./commands/update/UpdateCmd.js";
export * from "./constants/index.js";
export * from "./fn/exec.js";
export * from "./fn/render.js";
export * from "./interfaces/index.js";
export * from "./pipes/index.js";
export * from "./runtimes/index.js";
export * from "./services/CliProjectService.js";
export * from "./services/CliTemplatesService.js";
export * from "./services/ProjectClient.js";
export * from "./utils/defineTemplate.js";
