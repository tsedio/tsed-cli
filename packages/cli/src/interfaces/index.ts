export * from "./AlterGenerateTasks.js";
export * from "./AlterInitSubTasks.js";
export * from "./AlterPackageJson.js";
export * from "./AlterProjectFiles.js";
export * from "./AlterRenderFiles.js";
export * from "./ArchitectureConvention.js";
export * from "./CliCommandHooks.js";
export * from "./GenerateCmdContext.js";
export * from "./InitCmdOptions.js";
export * from "./PlatformType.js";
export * from "./ProjectConvention.js";
export * from "./RenderDataContext.js";
