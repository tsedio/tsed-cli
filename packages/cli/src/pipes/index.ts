export * from "./OutputFilePathPipe.js";
export * from "./RoutePipe.js";
export * from "./SymbolNamePipe.js";
