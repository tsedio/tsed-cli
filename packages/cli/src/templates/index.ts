// Import all templates to register them with the DI container
import "./asyncFactory.template.js";
import "./command.template.js";
import "./controller.template.js";
import "./decorator.template.js";
import "./exception-filter.template.js";
import "./factory.template.js";
import "./interface.template.js";
import "./interceptor.template.js";
import "./middleware.template.js";
import "./model.template.js";
import "./module.template.js";
import "./pipe.template.js";
import "./prisma.service.template.js";
import "./repository.template.js";
import "./response-filter.template.js";
import "./service.template.js";
import "./value.template.js";
import "./barrels.template.js";
import "./server.template.js";
import "./config.template.js";
import "./index.template.js";
import "./index.controller.template.js";
import "./index.command.template.js";
import "./index.logger.template.js";
import "./index.config.utils.template.js";
import "./tsconfig.template.js";
import "./tsconfig.spec.template.js";
import "./docker-compose.template.js";
import "./dockerfile.template.js";
import "./pm2.template.js";
import "./readme.template.js";
