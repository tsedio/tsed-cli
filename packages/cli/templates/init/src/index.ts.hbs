import {$log} from "@tsed/common";
import { {{platformSymbol}} } from "@tsed/platform-{{platform}}";
import {Server} from "./Server";

async function bootstrap() {
  try {
    const platform = await {{platformSymbol}}.bootstrap(Server);
    await platform.listen();

    process.on("SIGINT", () => {
      platform.stop();
    });
  } catch (error) {
    $log.error({event: "SERVER_BOOTSTRAP_ERROR", error});
  }
}

bootstrap();
