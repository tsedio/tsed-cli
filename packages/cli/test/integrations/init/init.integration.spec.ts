import {CliService, PackageManager, ProjectPackageJson} from "@tsed/cli-core";
import {CliPlatformTest, FakeCliFs} from "@tsed/cli-testing";
import {ensureDirSync, existsSync, readFileSync, writeFileSync} from "fs-extra";
import {dirname, join} from "path";
import {InitCmd, ProjectConvention, TEMPLATE_DIR} from "../../../src";
import filedirname from "filedirname";

const [, dir] = filedirname();

function readFile(file: string, content: string, rewrite = true) {
  const path = `${dir}/${file}`

  ensureDirSync(dirname(path))

  if (!existsSync(path) || rewrite) {
    writeFileSync(path, content, {encoding: "utf8"});
  }

  return readFileSync(path, {encoding: "utf8"});
}

describe("Init cmd", () => {
  beforeEach(() => {
      return CliPlatformTest.bootstrap({
        templateDir: TEMPLATE_DIR,
        commands: [InitCmd]
      });
    }
  );
  afterEach(() => CliPlatformTest.reset());

  describe("Express.js", () => {
    it("should generate a project with the right options", async () => {
      const cliService = CliPlatformTest.get<CliService>(CliService);
      const projectPackageJson = CliPlatformTest.get<ProjectPackageJson>(ProjectPackageJson);

      jest.spyOn(projectPackageJson as any, "getPackageJson").mockReturnValue({
        dependencies: {},
        devDependencies: {},
        scripts: {}
      });

      projectPackageJson.setRaw({
        name: "",
        version: "1.0.0",
        description: "",
        scripts: {},
        dependencies: {},
        devDependencies: {}
      });

      await cliService.exec("init", {
        platform: "express",
        rootDir: "./project-data",
        projectName: "project-data",
        tsedVersion: "5.58.1"
      });

      expect(FakeCliFs.getKeys()).toEqual([
        "./project-name",
        "project-name",
        "project-name/.barrelsby.json",
        "project-name/.dockerignore",
        "project-name/.gitignore",
        "project-name/Dockerfile",
        "project-name/README.md",
        "project-name/docker-compose.yml",
        "project-name/package.json",
        "project-name/processes.config.js",
        "project-name/src",
        "project-name/src/Server.ts",
        "project-name/src/config",
        "project-name/src/config/envs",
        "project-name/src/config/envs/index.ts",
        "project-name/src/config/index.ts",
        "project-name/src/config/logger",
        "project-name/src/config/logger/index.ts",
        "project-name/src/controllers/rest",
        "project-name/src/controllers/rest/HelloWorldController.ts",
        "project-name/src/index.ts",
        "project-name/tsconfig.compile.json",
        "project-name/tsconfig.json"
      ]);

      const content = FakeCliFs.entries.get("project-name/src/Server.ts")!;
      expect(content).toContain("import {Configuration, Inject} from \"@tsed/di\"");
      expect(content).toContain("import \"@tsed/platform-express\"");
      expect(content).toContain("import \"@tsed/ajv\"");
      expect(content).toEqual(readFile("data/Server.express.ts.txt", content));

      const pkg = JSON.parse(FakeCliFs.entries.get("project-name/package.json")!);
      expect(pkg).toEqual({
        "dependencies": {
          "@tsed/ajv": "5.58.1",
          "@tsed/common": "5.58.1",
          "@tsed/core": "5.58.1",
          "@tsed/di": "5.58.1",
          "@tsed/exceptions": "5.58.1",
          "@tsed/json-mapper": "5.58.1",
          "@tsed/platform-cache": "5.58.1",
          "@tsed/platform-exceptions": "5.58.1",
          "@tsed/platform-express": "5.58.1",
          "@tsed/platform-log-middleware": "5.58.1",
          "@tsed/platform-middlewares": "5.58.1",
          "@tsed/platform-params": "5.58.1",
          "@tsed/platform-response-filter": "5.58.1",
          "@tsed/platform-views": "5.58.1",
          "@tsed/schema": "5.58.1"
        },
        "description": "",
        "devDependencies": {},
        "name": "project-data",
        "scripts": {
          "barrels": "barrelsby --config .barrelsby.json",
          "build": "yarn run barrels && tsc --project tsconfig.compile.json",
          "start": "yarn run barrels && tsnd --inspect --ignore-watch node_modules --respawn --transpile-only -r tsconfig-paths/register src/index.ts",
          "start:prod": "cross-env NODE_ENV=production node dist/index.js"
        },
        "tsed": {
          "packageManager": "yarn"
        },
        "version": "1.0.0"
      });
    });
    it("should generate a project with swagger", async () => {
      const cliService = CliPlatformTest.get<CliService>(CliService);
      const projectPackageJson = CliPlatformTest.get<ProjectPackageJson>(ProjectPackageJson);

      jest.spyOn(projectPackageJson as any, "getPackageJson").mockReturnValue({
        dependencies: {},
        devDependencies: {},
        scripts: {}
      });

      projectPackageJson.setRaw({
        name: "",
        version: "1.0.0",
        description: "",
        scripts: {},
        dependencies: {},
        devDependencies: {}
      });

      await cliService.exec("init", {
        platform: "express",
        rootDir: "./project-data",
        projectName: "project-data",
        tsedVersion: "5.58.1",
        swagger: true
      });

      expect(FakeCliFs.getKeys()).toEqual([
        "./project-name",
        "project-name",
        "project-name/.barrelsby.json",
        "project-name/.dockerignore",
        "project-name/.gitignore",
        "project-name/Dockerfile",
        "project-name/README.md",
        "project-name/docker-compose.yml",
        "project-name/package.json",
        "project-name/processes.config.js",
        "project-name/src",
        "project-name/src/Server.ts",
        "project-name/src/config",
        "project-name/src/config/envs",
        "project-name/src/config/envs/index.ts",
        "project-name/src/config/index.ts",
        "project-name/src/config/logger",
        "project-name/src/config/logger/index.ts",
        "project-name/src/controllers/pages",
        "project-name/src/controllers/pages/IndexController.ts",
        "project-name/src/controllers/rest",
        "project-name/src/controllers/rest/HelloWorldController.ts",
        "project-name/src/index.ts",
        "project-name/tsconfig.compile.json",
        "project-name/tsconfig.json",
        "project-name/views",
        "project-name/views/swagger.ejs"
      ]);

      const content = FakeCliFs.entries.get("project-name/src/Server.ts")!;

      expect(content).toContain("import {Configuration, Inject} from \"@tsed/di\"");
      expect(content).toContain("import \"@tsed/platform-express\"");
      expect(content).toContain("import \"@tsed/ajv\"");
      expect(content).toContain("import * as pages from \"./controllers/pages/index\"");
      expect(content).toEqual(readFile("data/Server.express.swagger.ts.txt", content));

      const pkg = JSON.parse(FakeCliFs.entries.get("project-name/package.json")!);
      expect(pkg).toEqual({
        "dependencies": {
          "@tsed/ajv": "5.58.1",
          "@tsed/common": "5.58.1",
          "@tsed/core": "5.58.1",
          "@tsed/di": "5.58.1",
          "@tsed/exceptions": "5.58.1",
          "@tsed/json-mapper": "5.58.1",
          "@tsed/platform-cache": "5.58.1",
          "@tsed/platform-exceptions": "5.58.1",
          "@tsed/platform-express": "5.58.1",
          "@tsed/platform-log-middleware": "5.58.1",
          "@tsed/platform-middlewares": "5.58.1",
          "@tsed/platform-params": "5.58.1",
          "@tsed/platform-response-filter": "5.58.1",
          "@tsed/platform-views": "5.58.1",
          "@tsed/schema": "5.58.1"
        },
        "description": "",
        "devDependencies": {},
        "name": "project-data",
        "scripts": {
          "barrels": "barrelsby --config .barrelsby.json",
          "build": "yarn run barrels && tsc --project tsconfig.compile.json",
          "start": "yarn run barrels && tsnd --inspect --ignore-watch node_modules --respawn --transpile-only -r tsconfig-paths/register src/index.ts",
          "start:prod": "cross-env NODE_ENV=production node dist/index.js"
        },
        "tsed": {
          "packageManager": "yarn"
        },
        "version": "1.0.0"
      });
    });
    it("should generate a project with NPM", async () => {
      const cliService = CliPlatformTest.get<CliService>(CliService);
      const projectPackageJson = CliPlatformTest.get<ProjectPackageJson>(ProjectPackageJson);

      jest.spyOn(projectPackageJson as any, "getPackageJson").mockReturnValue({
        dependencies: {},
        devDependencies: {},
        scripts: {}
      });

      projectPackageJson.setRaw({
        name: "",
        version: "1.0.0",
        description: "",
        scripts: {},
        dependencies: {},
        devDependencies: {}
      });

      await cliService.exec("init", {
        platform: "express",
        rootDir: "./project-data",
        projectName: "project-data",
        tsedVersion: "5.58.1",
        packageManager: PackageManager.NPM
      });

      expect(FakeCliFs.getKeys()).toEqual([
        "./project-name",
        "project-name",
        "project-name/.barrelsby.json",
        "project-name/.dockerignore",
        "project-name/.gitignore",
        "project-name/Dockerfile",
        "project-name/README.md",
        "project-name/docker-compose.yml",
        "project-name/package.json",
        "project-name/processes.config.js",
        "project-name/src",
        "project-name/src/Server.ts",
        "project-name/src/config",
        "project-name/src/config/envs",
        "project-name/src/config/envs/index.ts",
        "project-name/src/config/index.ts",
        "project-name/src/config/logger",
        "project-name/src/config/logger/index.ts",
        "project-name/src/controllers/rest",
        "project-name/src/controllers/rest/HelloWorldController.ts",
        "project-name/src/index.ts",
        "project-name/tsconfig.compile.json",
        "project-name/tsconfig.json"
      ]);

      const content = FakeCliFs.entries.get("project-name/src/Server.ts")!;
      expect(content).toContain("import {Configuration, Inject} from \"@tsed/di\"");
      expect(content).toContain("import \"@tsed/platform-express\"");
      expect(content).toContain("import \"@tsed/ajv\"");
      expect(content).toEqual(readFile("data/Server.express.ts.txt", content));

      const pkg = JSON.parse(FakeCliFs.entries.get("project-name/package.json")!);
      expect(pkg).toEqual({
        "dependencies": {
          "@tsed/ajv": "5.58.1",
          "@tsed/common": "5.58.1",
          "@tsed/core": "5.58.1",
          "@tsed/di": "5.58.1",
          "@tsed/exceptions": "5.58.1",
          "@tsed/json-mapper": "5.58.1",
          "@tsed/platform-cache": "5.58.1",
          "@tsed/platform-exceptions": "5.58.1",
          "@tsed/platform-express": "5.58.1",
          "@tsed/platform-log-middleware": "5.58.1",
          "@tsed/platform-middlewares": "5.58.1",
          "@tsed/platform-params": "5.58.1",
          "@tsed/platform-response-filter": "5.58.1",
          "@tsed/platform-views": "5.58.1",
          "@tsed/schema": "5.58.1"
        },
        "description": "",
        "devDependencies": {},
        "name": "project-data",
        "scripts": {
          "barrels": "barrelsby --config .barrelsby.json",
          "build": "npm run barrels && tsc --project tsconfig.compile.json",
          "start": "npm run barrels && tsnd --inspect --ignore-watch node_modules --respawn --transpile-only -r tsconfig-paths/register src/index.ts",
          "start:prod": "cross-env NODE_ENV=production node dist/index.js"
        },
        "tsed": {
          "packageManager": "npm"
        },
        "version": "1.0.0"
      });
    });
    it("should generate a project with Convention ANGULAR", async () => {
      const cliService = CliPlatformTest.get<CliService>(CliService);
      const projectPackageJson = CliPlatformTest.get<ProjectPackageJson>(ProjectPackageJson);

      jest.spyOn(projectPackageJson as any, "getPackageJson").mockReturnValue({
        dependencies: {},
        devDependencies: {},
        scripts: {}
      });

      projectPackageJson.setRaw = (pkg) => {
        // @ts-ignore
        projectPackageJson.raw = {
          name: "",
          version: "1.0.0",
          description: "",
          scripts: {},
          dependencies: {},
          devDependencies: {},
          tsed: {
            convention: "angular"
          }
        }
      };

      await cliService.exec("init", {
        platform: "express",
        rootDir: "./project-data",
        projectName: "project-data",
        tsedVersion: "5.58.1",
        convention: ProjectConvention.ANGULAR,
        swagger: true
      });

      expect(FakeCliFs.getKeys()).toEqual([
        "./project-name",
        "project-name",
        "project-name/.barrelsby.json",
        "project-name/.dockerignore",
        "project-name/.gitignore",
        "project-name/Dockerfile",
        "project-name/README.md",
        "project-name/docker-compose.yml",
        "project-name/package.json",
        "project-name/processes.config.js",
        "project-name/src",
        "project-name/src/config",
        "project-name/src/config/envs",
        "project-name/src/config/envs/index.ts",
        "project-name/src/config/index.ts",
        "project-name/src/config/logger",
        "project-name/src/config/logger/index.ts",
        "project-name/src/controllers/pages",
        "project-name/src/controllers/pages/index.controller.ts",
        "project-name/src/controllers/rest",
        "project-name/src/controllers/rest/hello-world.controller.ts",
        "project-name/src/index.ts",
        "project-name/src/server.ts",
        "project-name/tsconfig.compile.json",
        "project-name/tsconfig.json",
        "project-name/views",
        "project-name/views/swagger.ejs"
      ]);

      const content = FakeCliFs.entries.get("project-name/src/server.ts")!;
      expect(content).toContain("import {Configuration, Inject} from \"@tsed/di\"");
      expect(content).toContain("import \"@tsed/platform-express\"");
      expect(content).toContain("import \"@tsed/ajv\"");
      expect(content).toEqual(readFile("data/Server.express.ts.txt", content));
      expect(content).toContain("import * as pages from \"./controllers/pages/index\"");
      expect(content).toContain("export class Server {");
    });
  });

  describe("Koa.js", () => {
    it("should generate a project with the right options", async () => {
      const cliService = CliPlatformTest.get<CliService>(CliService);
      const projectPackageJson = CliPlatformTest.get<ProjectPackageJson>(ProjectPackageJson);

      jest.spyOn(projectPackageJson as any, "getPackageJson").mockReturnValue({
        dependencies: {},
        devDependencies: {},
        scripts: {}
      });

      projectPackageJson.setRaw({
        name: "",
        version: "1.0.0",
        description: "",
        scripts: {},
        dependencies: {},
        devDependencies: {}
      });

      await cliService.exec("init", {
        platform: "koa",
        rootDir: "./project-data",
        projectName: "project-data",
        tsedVersion: "5.58.1"
      });

      expect(FakeCliFs.getKeys()).toEqual([
        "./project-name",
        "project-name",
        "project-name/.barrelsby.json",
        "project-name/.dockerignore",
        "project-name/.gitignore",
        "project-name/Dockerfile",
        "project-name/README.md",
        "project-name/docker-compose.yml",
        "project-name/package.json",
        "project-name/processes.config.js",
        "project-name/src",
        "project-name/src/Server.ts",
        "project-name/src/config",
        "project-name/src/config/envs",
        "project-name/src/config/envs/index.ts",
        "project-name/src/config/index.ts",
        "project-name/src/config/logger",
        "project-name/src/config/logger/index.ts",
        "project-name/src/controllers/rest",
        "project-name/src/controllers/rest/HelloWorldController.ts",
        "project-name/src/index.ts",
        "project-name/tsconfig.compile.json",
        "project-name/tsconfig.json"
      ]);

      const content = FakeCliFs.entries.get("project-name/src/Server.ts")!;

      expect(content).toContain("import {Configuration, Inject} from \"@tsed/di\"");
      expect(content).toContain("import \"@tsed/platform-koa\"");
      expect(content).toContain("import \"@tsed/ajv\"");
      expect(content).toEqual(readFile("data/Server.koa.ts.txt", content));

      const pkg = JSON.parse(FakeCliFs.entries.get("project-name/package.json")!);
      expect(pkg).toEqual({
        "dependencies": {
          "@tsed/ajv": "5.58.1",
          "@tsed/common": "5.58.1",
          "@tsed/core": "5.58.1",
          "@tsed/di": "5.58.1",
          "@tsed/exceptions": "5.58.1",
          "@tsed/json-mapper": "5.58.1",
          "@tsed/platform-cache": "5.58.1",
          "@tsed/platform-exceptions": "5.58.1",
          "@tsed/platform-koa": "5.58.1",
          "@tsed/platform-log-middleware": "5.58.1",
          "@tsed/platform-middlewares": "5.58.1",
          "@tsed/platform-params": "5.58.1",
          "@tsed/platform-response-filter": "5.58.1",
          "@tsed/platform-views": "5.58.1",
          "@tsed/schema": "5.58.1"
        },
        "description": "",
        "devDependencies": {},
        "name": "project-data",
        "scripts": {
          "barrels": "barrelsby --config .barrelsby.json",
          "build": "yarn run barrels && tsc --project tsconfig.compile.json",
          "start": "yarn run barrels && tsnd --inspect --ignore-watch node_modules --respawn --transpile-only -r tsconfig-paths/register src/index.ts",
          "start:prod": "cross-env NODE_ENV=production node dist/index.js"
        },
        "tsed": {
          "packageManager": "yarn"
        },
        "version": "1.0.0"
      });
    });
  });

  describe("shared configuration", () => {
    it("should configuration directory", async () => {
      const cliService = CliPlatformTest.get<CliService>(CliService);
      const projectPackageJson = CliPlatformTest.get<ProjectPackageJson>(ProjectPackageJson);

      jest.spyOn(projectPackageJson as any, "getPackageJson").mockReturnValue({
        dependencies: {},
        devDependencies: {},
        scripts: {}
      });

      projectPackageJson.setRaw({
        name: "",
        version: "1.0.0",
        description: "",
        scripts: {},
        dependencies: {},
        devDependencies: {}
      });

      await cliService.exec("init", {
        platform: "express",
        rootDir: "./project-data",
        projectName: "project-data",
        tsedVersion: "5.58.1",
        commands: true
      });

      try {
        FakeCliFs.getKeys().map((key: string) => {
          const content = FakeCliFs.entries.get(key)!

          if (content !== key) {
            writeFileSync(join(dir, "data", key), content, {encoding: "utf-8"})
          } else {
            ensureDirSync(join(dir, "data", key))
          }
        })
      } catch (er) {
      }

      expect(FakeCliFs.getKeys()).toEqual([
        "./project-name",
        "project-name",
        "project-name/.barrelsby.json",
        "project-name/.dockerignore",
        "project-name/.gitignore",
        "project-name/Dockerfile",
        "project-name/README.md",
        "project-name/docker-compose.yml",
        "project-name/package.json",
        "project-name/processes.config.js",
        "project-name/src",
        "project-name/src/Server.ts",
        "project-name/src/bin",
        "project-name/src/bin/HelloCommand.ts",
        "project-name/src/bin/index.ts",
        "project-name/src/config",
        "project-name/src/config/envs",
        "project-name/src/config/envs/index.ts",
        "project-name/src/config/index.ts",
        "project-name/src/config/logger",
        "project-name/src/config/logger/index.ts",
        "project-name/src/controllers/rest",
        "project-name/src/controllers/rest/HelloWorldController.ts",
        "project-name/src/index.ts",
        "project-name/tsconfig.compile.json",
        "project-name/tsconfig.json"
      ]);
    });
  })
});
