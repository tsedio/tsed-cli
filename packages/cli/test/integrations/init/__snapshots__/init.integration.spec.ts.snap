// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Init cmd > Express.js > should generate a project with Arch FEATURE 2`] = `
"import "@tsed/ajv";
import "@tsed/platform-log-request";

import { Configuration } from "@tsed/di";
import { application } from "@tsed/platform-http";
import { join } from "node:path";

import { config } from "@/config/config.js";
import "@tsed/platform-express";
import "@tsed/swagger";
import * as pages from "./pages/index.js";
import * as rest from "./rest/index.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [...Object.values(rest), ...Object.values(rest)],
    "/": [...Object.values(pages), ...Object.values(pages)]
  },
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  },
  swagger: [
    {
      "path": "/doc",
      "specVersion": "3.1.0"
    }
  ],
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    {
      "use": "urlencoded-parser",
      "options": {
        "extended": true
      }
    }
  ]
})
export class Server {
  protected app = application();
}
"
`;

exports[`Init cmd > Express.js > should generate a project with Babel 2`] = `
"import "@tsed/ajv";
import "@tsed/platform-log-request";

import { Configuration } from "@tsed/di";
import { application } from "@tsed/platform-http";
import { join } from "node:path";

import { config } from "@/config/config.js";
import "@tsed/platform-express";
import * as rest from "./controllers/rest/index.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [...Object.values(rest), ...Object.values(rest)]
  },
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  },
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    {
      "use": "urlencoded-parser",
      "options": {
        "extended": true
      }
    }
  ]
})
export class Server {
  protected app = application();
}
"
`;

exports[`Init cmd > Express.js > should generate a project with Bun 2`] = `
"import "@tsed/ajv";
import "@tsed/platform-log-request";

import { Configuration } from "@tsed/di";
import { application } from "@tsed/platform-http";
import { join } from "node:path";

import { config } from "@/config/config.js";
import "@tsed/platform-express";
import * as rest from "./controllers/rest/index.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [...Object.values(rest), ...Object.values(rest)]
  },
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  },
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    {
      "use": "urlencoded-parser",
      "options": {
        "extended": true
      }
    }
  ]
})
export class Server {
  protected app = application();
}
"
`;

exports[`Init cmd > Express.js > should generate a project with Convention ANGULAR 2`] = `
"import "@tsed/ajv";
import "@tsed/platform-log-request";

import { Configuration } from "@tsed/di";
import { application } from "@tsed/platform-http";
import { join } from "node:path";

import { config } from "@/config/config.js";
import "@tsed/platform-express";
import "@tsed/swagger";
import * as pages from "./controllers/pages/index.js";
import * as rest from "./controllers/rest/index.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [...Object.values(rest), ...Object.values(rest)],
    "/": [...Object.values(pages), ...Object.values(pages)]
  },
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  },
  swagger: [
    {
      "path": "/doc",
      "specVersion": "3.1.0"
    }
  ],
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    {
      "use": "urlencoded-parser",
      "options": {
        "extended": true
      }
    }
  ]
})
export class Server {
  protected app = application();
}
"
`;

exports[`Init cmd > Express.js > should generate a project with NPM 2`] = `
"import "@tsed/ajv";
import "@tsed/platform-log-request";

import { Configuration } from "@tsed/di";
import { application } from "@tsed/platform-http";
import { join } from "node:path";

import { config } from "@/config/config.js";
import "@tsed/platform-express";
import * as rest from "./controllers/rest/index.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [...Object.values(rest), ...Object.values(rest)]
  },
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  },
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    {
      "use": "urlencoded-parser",
      "options": {
        "extended": true
      }
    }
  ]
})
export class Server {
  protected app = application();
}
"
`;

exports[`Init cmd > Express.js > should generate a project with Webpack 2`] = `
"import "@tsed/ajv";
import "@tsed/platform-log-request";

import { Configuration } from "@tsed/di";
import { application } from "@tsed/platform-http";
import { join } from "node:path";

import { config } from "@/config/config.js";
import "@tsed/platform-express";
import * as rest from "./controllers/rest/index.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [...Object.values(rest), ...Object.values(rest)]
  },
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  },
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    {
      "use": "urlencoded-parser",
      "options": {
        "extended": true
      }
    }
  ]
})
export class Server {
  protected app = application();
}
"
`;

exports[`Init cmd > Express.js > should generate a project with swagger 3`] = `
"import "@tsed/ajv";
import "@tsed/platform-log-request";

import { Configuration } from "@tsed/di";
import { application } from "@tsed/platform-http";
import { join } from "node:path";

import { config } from "@/config/config.js";
import "@tsed/platform-express";
import "@tsed/swagger";
import * as pages from "./controllers/pages/index.js";
import * as rest from "./controllers/rest/index.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [...Object.values(rest), ...Object.values(rest)],
    "/": [...Object.values(pages), ...Object.values(pages)]
  },
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  },
  swagger: [
    {
      "path": "/doc",
      "specVersion": "3.1.0"
    }
  ],
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    {
      "use": "urlencoded-parser",
      "options": {
        "extended": true
      }
    }
  ]
})
export class Server {
  protected app = application();
}
"
`;

exports[`Init cmd > Express.js > should generate a project with the right options 2`] = `
"import "@tsed/ajv";
import "@tsed/platform-log-request";

import { Configuration } from "@tsed/di";
import { application } from "@tsed/platform-http";
import { join } from "node:path";

import { config } from "@/config/config.js";
import "@tsed/platform-express";
import * as rest from "./controllers/rest/index.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [...Object.values(rest), ...Object.values(rest)]
  },
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  },
  middlewares: [
    "cors",
    "cookie-parser",
    "compression",
    "method-override",
    "json-parser",
    {
      "use": "urlencoded-parser",
      "options": {
        "extended": true
      }
    }
  ]
})
export class Server {
  protected app = application();
}
"
`;

exports[`Init cmd > Fastify.js > should generate a project with the right options 2`] = `
"import "@tsed/ajv";
import "@tsed/platform-log-request";

import { Configuration } from "@tsed/di";
import { application } from "@tsed/platform-http";
import { join } from "node:path";

import { config } from "@/config/config.js";
import "@tsed/platform-fastify";
import * as rest from "./controllers/rest/index.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [...Object.values(rest), ...Object.values(rest)]
  },
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  },
  plugins: [
    "@fastify/accepts",
    "@fastify/cookie",
    {
      "use": "fastify-raw-body",
      "options": {
        "global": false,
        "runFirst": true
      }
    },
    "@fastify/formbody"
  ]
})
export class Server {
  protected app = application();
}
"
`;

exports[`Init cmd > Koa.js > should generate a project with the right options 2`] = `
"import "@tsed/ajv";
import "@tsed/platform-log-request";

import { Configuration } from "@tsed/di";
import { application } from "@tsed/platform-http";
import { join } from "node:path";

import { config } from "@/config/config.js";
import "@tsed/platform-koa";
import * as rest from "./controllers/rest/index.js";

@Configuration({
  ...config,
  acceptMimes: ["application/json"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  mount: {
    "/rest": [...Object.values(rest), ...Object.values(rest)]
  },
  views: {
    root: join(process.cwd(), "../views"),
    extensions: {
      ejs: "ejs"
    }
  },
  middlewares: [
    "@koa/cors",
    "koa-compress",
    "koa-override",
    "koa-bodyparser"
  ]
})
export class Server {
  protected app = application();
}
"
`;
